src/core/thread/mesh_forwarder.cpp

```diff
@@ -1419,31 +1419,26 @@ void MeshForwarder::HandleFragment(FrameData            &aFrameData,
     {
         Neighbor *neighbor = Get<NeighborTable>().FindNeighbor(aMacAddrs.mSource, Neighbor::kInStateAnyExceptInvalid);
 
-        if (neighbor != nullptr)
+        if ((neighbor != nullptr) && (fragmentHeader.GetDatagramOffset() == 0))
```

Comment:
Ideally in case the tag matches previously received tag, then we can accept the fragments whose offset > previously received offset, else drop? In the current approach, we would accept even if the offset is duplicate of previously received offset?

Reply:
For any "next fragment"  we don't drop it here in this part of code, but there is code below in the same method that would check the offset (with what we have in reassembly list) and drops if the offset is not correct (e.g. if offset is same as what we received last).

This comment in the code explains this more:
```
        // Duplication suppression for a "next fragment" is handled
        // by the code below where the the datagram offset is
        // checked against the offset of the corresponding message
        // (same datagram tag and size) in Reassembly List. Note
        // that if there is no matching message in the Reassembly
        // List (e.g., in case the message is already fully
        // assembled) the received "next fragment" frame would be
        // dropped.
```

Reply:
thanks for the clarification, looks good.